python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -98,8 +99,15 @@ def to_pandas_dtype(self):
 
         return TensorDtype(self._shape, self.storage_type.value_type.to_pandas_dtype())
 
-    def __reduce__(self):
-        return ArrowTensorType, (self._shape, self.storage_type.value_type)
+    def __arrow_ext_serialize__(self):
+        arguments = (self._shape, self.storage_type.value_type)
+        return pickle.dumps(arguments)
```

Comment:
You might want to use a custom serialization format instead of pickle, because in theory this allows someone to manually create a Parquet or Feather/IPC file with the "ray.data.arrow_tensor" extension name but with malicious metadata. 
For the canonical extension types we have in Arrow itself we use JSON (for example, see "description of the serialization" section at https://arrow.apache.org/docs/format/CanonicalExtensions.html#fixed-shape-tensor, or an example in python in the pandas code base: https://github.com/pandas-dev/pandas/blob/851fea0ea38985cd7d5e0a3b7a7a7539b5883307/pandas/core/arrays/arrow/extension_types.py#L32-L39), or in the docs a simple custom string format is used (https://arrow.apache.org/docs/python/extending_types.html#parameterized-extension-type)

Reply:
@jorisvandenbossche ah, that's good know. Thanks for the pointer!